spec/services/deadline_calculator/calendar_month_spec.rb

```diff
@@ -0,0 +1,117 @@
+        expected_deadline = expected_deadline.tomorrow
+    end
+    expected_deadline
+end
+
+describe DeadlineCalculator::CalendarMonth do
+    let(:sar)                 { find_or_create :sar_correspondence_type }
+    let(:sar_case)            { create :sar_case }
+    let(:deadline_calculator) { described_class.new sar_case }
+  
+    context 'SAR case' do
+      describe '#escalation deadline' do
+        it 'is 3 calendar days after the date of creation' do
+          expect(deadline_calculator.escalation_deadline)
+            .to eq 3.days.since(sar_case.created_at.to_date)
+        end
+      end
+  
+      describe '#internal deadline' do
+        it 'is 10 calendar days after the date received' do
+          expect(deadline_calculator.internal_deadline)
+            .to eq 10.days.since(sar_case.received_date)
+        end
+      end
+  
+      describe '#internal_deadline_for_date' do
+        it 'is 10 calendar days from the supplied date' do
+          expect(deadline_calculator.internal_deadline_for_date(sar, Date.new(2018, 6, 25))).to eq Date.new(2018, 7, 5)
+        end
+      end
+      
+      
+      describe '#external deadline' do
+        context 'general cases for deadlines' do
+            dates_for_testing = [
+              {"received_date" => { "month" => 1, "day" => 31 }, "expected_base_date" => { "month" => 2, "day" => -1 } }, 
+              {"received_date" => { "month" => 6, "day" => 30 }, "expected_base_date" => { "month" => 7, "day" => 30 } }, 
+              {"received_date" => { "month" => 2, "day" => -1 }, "expected_base_date" => { "month" => 3, "day" => 0 } }, 
+              {"received_date" => { "month" => 3, "day" => 15 }, "expected_base_date" => { "month" => 4, "day" => 15 } }, 
+            ]
+            dates_for_testing.each do |date_for_testing|
+              month = date_for_testing["received_date"]["month"]
+              day = date_for_testing["received_date"]["day"]
+              base_month = date_for_testing["expected_base_date"]["month"]
+              which_year, received_date_used = generate_start_date(month, day)
+              base_day = date_for_testing["expected_base_date"]["day"] 
+              if base_day == 0 
+                base_day = received_date_used.day
+              end
+              expected_deadline = get_expected_deadline(Date.civil(which_year, base_month, base_day))
+              it 'Testing #{received_date_used} - #{expected_deadline}' do
+                test_case = double('sar_case')
+                allow(test_case).to receive(:received_date).and_return(received_date_used)
+                deadline_calculator_local = described_class.new test_case
+                expect(deadline_calculator_local.external_deadline).to eq expected_deadline
+              end
+            end
+        end
+
+        context 'the date of one calendar month later is non working date' do
+          it 'weekend ' do
```

Comment:
I wonder if we could add a bit more context for the descriptions, to make it clearer what the business rules are and why this class exists. 

For something like this, it's really good to run the specs with the `--documentation` flag on and make sure it generates readable sentences... This is one bit we really need the tests be our documentation :) 

Maybe a comment block in the CalendarMonth class too?

Reply:
Added some documents on CalendarMonths,  will add more description here